src/ansi.rs

```diff
@@ -489,58 +513,51 @@ impl From<char> for OpChar {
     }
 }
 
-pub struct TerminalEsc {
-    buffer: String,
+pub fn parse_esc_str(s: &str) -> ParseRes {
+    parse_esc_str_tail(s, vec![])
 }
 
 ///
-/// Future buffered version
+/// buffered version
 /// "hello" -> "hello" (nom returns Error)
 /// "ESC[","garbage" -> `InSequence`, "arbage" (nom returns Failure)
-/// "ESC[Ablah", "garbage" -> [Foo(Op), Foo("blah")], [Foo("garbage")],
+/// "ESC[Ablah", "garbage" -> [Foo(Op(A)), Foo("blah")], [Foo("garbage")],
+/// "garbageESC[Ablah" -> ["garbage", Op(A), "blah"]
+/// "garbageESC[Xblah" -> ["garbage", "blah"]
```

Comment:
It seems to me that we've actually implemented:

```suggestion
/// "garbageESC[Xblah" -> ["garbage", "[Xblah"]
```

Reply:
Do we not eat the first char to break a sequence?

Reply:
Do we? It looks to me like we're only eating one char (the ESC) in the failure case

Reply:
We are eating 1 char but I thought at the time it didn't include any of the successful bytes I think? Am I/the code wrong about that?